worker/echo/main.go

```diff
@@ -12,16 +12,23 @@ import (
 
 	"git.sr.ht/~spc/go-log"
 
+	"github.com/redhatinsights/yggdrasil/internal/sync"
```

Comment:
This is okay because the echo worker is part of the yggdrasil repository, but I
don't know if Go would permit this import from an external project. I think the
'internal' package has special value to the Go compiler. Might be worth testing.

We could easily solve this by moving the `sync` package to the top of the
`yggdrasil` project, but it seems weird for the `yggdrasil` package to export a
`sync` package.

Reply:
I think this sync package might be useful for worker development, in case other workers want to use this utilty.

Reply:
What do you think to move this pieces of code into the pkg directory?

```
yggdrasil
.
├── build-aux
├── builddir
├── cmd
├── data
├── dbus
├── dist
├── doc
├── internal
├── ipc
├── lib
├── pkg
└── worker
```

That way, we have such an standard way to do the same things. I'm thinking as well in the `http` implementation of the client we have in `yggdrasil`, it can be also be used in `rhc`.

Reply:
Yes, we could move `sync` out of `internal` and into the top level of the project. I would prefer to avoid creating a `pkg` package. It is unnecessary and creates an extraneous entry in import paths:

#### With `pkg`

```go
import "github.com/redhatinsights/yggdrasil/pkg/sync"
```

#### Without `pkg`
```go
import "github.com/redhatinsights/yggdrasil/sync"
```